geoutils/georaster.py

```diff
@@ -579,6 +579,20 @@ def reproject(self, dst_ref=None, dst_crs=None, dst_size=None, dst_bounds=None,
             dst_data = np.ones(dst_shape)
             reproj_kwargs.update({'destination': dst_data})
 
+        # Check that reprojection is actually needed
+        if all([
+                (dst_transform == self.transform) or (dst_transform is None),
+                (dst_crs == self.crs) or (dst_crs is None),
+                (dst_size == self.shape) or (dst_size is None),
+                (dst_res == self.res) or (dst_res == self.res[0] == self.res[1]) or (dst_res is None)
+        ]):
+            if (nodata == self.nodata) or (nodata is None):
+                warnings.warn("Output projection, bounds and size are identical -> return self")
```

Comment:
I put a warning for now. Is that the most sensible?

Reply:
Mmmh.. Don't know. We could even skip the warning?

Reply:
After thinking, I think there should be a warning. In all other cases, reproject returns a copy, here we return self, so any changes made to the output after reproject will also be applied to the original dataset...

Reply:
Ah yes, fair enough. Maybe write this in the warning (not returning a copy)

Reply:
Done